File: system-test/dotnet/Dsl.Core/UseCase/External/Erp/UseCases/GetProduct.cs
using Driver.Port.External.Erp;
using Driver.Port.External.Erp.Dtos;
using Driver.Port.External.Erp.Dtos.Error;
using Dsl.Core.External.Erp.UseCases.Base;
using Dsl.Core.Shared;

namespace Dsl.Core.External.Erp.UseCases;

public class GetProduct : BaseErpCommand<GetProductResponse, GetProductVerification>
{
    private string? _skuParamAlias;

    public GetProduct(IErpDriver driver, UseCaseContext context)
        : base(driver, context)
    {
    }

    public GetProduct Sku(string? skuParamAlias)
    {
        _skuParamAlias = skuParamAlias;
        return this;
    }

    public override async Task<ErpUseCaseResult<GetProductResponse, GetProductVerification>> Execute()
    {
        var sku = _context.GetParamValue(_skuParamAlias);

        var request = new GetProductRequest
        {
            Sku = sku
        };

        var result = await _driver.GetProductAsync(request);

        return new ErpUseCaseResult<GetProductResponse, GetProductVerification>(
            result,
            _context,
            (response, ctx) => new GetProductVerification(response, ctx));
    }
}

File: system-test/java/src/main/java/com/optivem/shop/testkit/core/usecase/external/erp/usecases/GetProduct.java
package com.optivem.shop.testkit.core.usecase.external.erp.usecases;

import com.optivem.shop.testkit.driver.port.external.erp.ErpDriver;
import com.optivem.shop.testkit.driver.port.external.erp.dtos.GetProductRequest;
import com.optivem.shop.testkit.driver.port.external.erp.dtos.GetProductResponse;
import com.optivem.shop.testkit.core.usecase.external.erp.usecases.base.BaseErpUseCase;
import com.optivem.shop.testkit.core.shared.UseCaseResult;
import com.optivem.shop.testkit.core.shared.UseCaseContext;

public class GetProduct extends BaseErpUseCase<GetProductResponse, GetProductVerification> {
    private String skuParamAlias;

    public GetProduct(ErpDriver driver, UseCaseContext context) {
        super(driver, context);
    }

    public GetProduct sku(String skuParamAlias) {
        this.skuParamAlias = skuParamAlias;
        return this;
    }

    @Override
    public UseCaseResult<GetProductResponse, GetProductVerification> execute() {
        var sku = context.getParamValue(skuParamAlias);

        var request = GetProductRequest.builder()
                .sku(sku)
                .build();

        var result = driver.getProduct(request);

        return new UseCaseResult<>(result, context, GetProductVerification::new);
    }
}
